app/services/transaction_service/paypal_events.rb

```diff
@@ -50,34 +62,66 @@ def update_transaction_details(flow, details)
 
   # Privates
 
+  def set_session_context!(actor:, tx:)
+    existing_session_ctx = SessionContextStore.get
+
+    marketplace_session_ctx = {
+      marketplace_id: tx[:community_id],
+      marketplace_uuid: tx[:community_uuid]
+    }
+
+    user_session_ctx =
+      case actor
+      when :starter
+        {
+          user_id: tx[:starter_id],
+          user_uuid: tx[:starter_uuid]
+        }
+      when :author
+        {
+          user_id: tx[:listing_author_id],
+          user_uuid: tx[:listing_author_uuid]
+        }
+      when :unknown
+        # Unknown user
+        {}
+      else
+        raise ArgumentError.new("Unknown transition actor: #{actor}")
+      end
```

Comment:
How about adding a helper method to SessionContextStore similar to set_from_models that would build a proper session context from given parameters? That way the details of which keys are used would remain internal to store.